js/src/containers/Twitter.js

```diff
@@ -0,0 +1,122 @@
+/* External dependencies */
+import { connect } from "react-redux";
+import { actions } from "@yoast/social-metadata-forms";
+
+/* Internal dependencies */
+import TwitterWrapper from "../components/social/TwitterWrapper";
+import { socialSelectors }  from "../redux/selectors/socialSelectors";
+import { getImageFallback, getDescriptionFallback, getTitleFallback } from "../redux/selectors/fallbackSelectors";
+
+const {
+	setSocialPreviewTitle,
+	setSocialPreviewDescription,
+	setSocialPreviewImage,
+	setSocialPreviewImageUrl,
+	setSocialPreviewImageId,
+} = actions;
+
+/**
+ * Maps the state to props.
+ *
+ * @param {Object} state The state.`
+ *
+ * @returns {Object} The props for the TwitterView component.
+ */
+const mapStateToProps = ( state ) => {
+	const image = state.socialReducer.twitter.image;
+	return {
+		recommendedReplacementVariables: state.settings.snippetEditor.recommendedReplacementVariables,
+		replacementVariables: state.snippetEditor.replacementVariables,
+		description: socialSelectors.getTwitterDescription( state ) || socialSelectors.getFacebookDescription( state ) || getDescriptionFallback( state ),
+		descriptionFallback: "Modify your Twitter description by editing it right here...",
+		title: socialSelectors.getTwitterTitle( state ),
+		titleFallback: socialSelectors.getFacebookTitle( state ) || getTitleFallback( state ),
+		image: { ...image, fallbackUrl: state.socialReducer.facebook.image.url || getImageFallback( state ) },
+		imageWarnings: state.socialReducer.twitter.warnings,
+	};
+};
+
+const titleInput = document.getElementById( "yoast_wpseo_twitter-title" );
+const descriptionInput = document.getElementById( "yoast_wpseo_twitter-description" );
+const imageIdInput = document.getElementById( "yoast_wpseo_twitter-image-id" );
+const imageUrlInput = document.getElementById( "yoast_wpseo_twitter-image" );
+
+// Make the media library accessible.
+const media = window.wp.media();
+
+// Listens for the selection of an image. Then gets the right data and dispatches the data to the store.
+media.on( "select", () => {
```

Comment:
This is _almost_ the same listener function that is registered on both the Twitter and on the Facebook container. You can extract it to some helper function location.

The difference, namely the `"twitter"` and `"facebook"` is only relevant to the dispatched function, and could be passed as a prop.

E.g. (note I also like more whitespace, as you would put paragraphs in a text. But that's personal preference).

```js
const registerMediaSelectListener = ( platform ) => {
	const media = window.wp.media();
	media.on( "select", () => {
		const selected = media.state().get( "selection" ).first();

		const {
			filesizeInBytes,
			subtype,
			height,
			width,
			url,
			id,
		} = selected.attributes;

		window.wp.data.dispatch( "yoast-seo/editor" ).setSocialPreviewImage( {
			bytes: filesizeInBytes,
			type: subtype,
			url,
			id,
			width,
			height,
		}, platform );
		imageIdInput.value = id;
		imageUrlInput.value = url;
	}
}
```
It can probably be abstracted way further, but I guess, for a first iteration it may suffice.

On each container you'd then only have to call `registerMediaSelectListener( "twitter" );` for example.